### PR TITLE
Feature/mdr 71

### PR DESCRIPTION
**Pull request by mclaurentiu**

### CODE REVIEW

File: validator/__init__.py [hunk 1]
```diff
@@ -5,4 +5,7 @@
 # Author: Eugeniu Costetchi
 # Email: costezki.eugen@gmail.com 
 
-""" """
\ No newline at end of file
+""" """
+import logging
+
+logging.basicConfig(filename='app.log', filemode='w', format='%(name)s - %(levelname)s - %(message)s')
```
Comment (costezki) [inline validator/__init__.py:11]: good practice

File: validator/service_layer/handlers.py [hunk 2]
```diff
@@ -27,7 +28,7 @@ def __copy_static_content(from_path, to_path):
         __logger.warning(from_path + " is not a directory !")
 
 
-def run_file_validator(dataset_uri: str, data_file: str, schemas: List[str], output: Path) -> str:
+def run_file_validator(dataset_uri: str, data_file: str, schemas: List[str], output: Path) -> tuple:
```
Comment (cosletmihai) [inline validator/service_layer/handlers.py:31]: update the `:return:` info in the method docs description

File: validator/service_layer/handlers.py [hunk 3]
```diff
@@ -47,52 +48,60 @@ def run_file_validator(dataset_uri: str, data_file: str, schemas: List[str], out
...
+#
+#     parsed_uri = urlparse(dataset_uri)
+#     output_file_name = parsed_uri.netloc.replace(":", "_") + \
+#                        parsed_uri.path.replace("/", "_") + \
+#                        ".shaclTestCaseResult.html"
+#
+#     return Path(output) / "results" / output_file_name
 
 
 def run_sparql_endpoint_validator(dataset_uri: str, sparql_endpoint_uri: str, graphs_uris: List[str],
```
Comment (cosletmihai) [inline validator/service_layer/handlers.py:103]: update the `:return:` info in the method docs description

File: validator/entrypoints/api/handlers.py [hunk 4]
```diff
@@ -8,15 +8,19 @@
...
+from flask import send_file
 from werkzeug.datastructures import FileStorage
 from werkzeug.exceptions import UnsupportedMediaType, InternalServerError
 
 from validator.entrypoints.api.helpers import _guess_file_type, INPUT_MIME_TYPES
-from validator.service_layer.handlers import run_file_validator, run_sparql_endpoint_validator
+from validator.service_layer.handlers import run_file_validator, run_sparql_endpoint_validator, \
+    prepare_eds4jinja_context, generate_validation_report
+
+__logger = logging.getLogger(__name__)
```
Comment (costezki) [inline validator/entrypoints/api/handlers.py:23]: I know we want to mark non-external things, but, at least for me, is easier to use the variable when the name is without underscores.

File: validator/service_layer/handlers.py [hunk 5]
```diff
@@ -47,52 +48,60 @@ def run_file_validator(dataset_uri: str, data_file: str, schemas: List[str], out
...
-                       ".shaclTestCaseResult.html"
-
-    return Path(output) / "results" / output_file_name
+    output_file_name = str(Path(data_file).parent).replace('/', '_') + '_' + Path(
+        data_file).name + ".shaclTestCaseResult"
+    output_file_full_path = Path(output) / 'results' / output_file_name
+    return str(output_file_full_path) + ".html", str(output_file_full_path) + ".ttl"
+
+
+# def run_endpoint_validator(dataset_uri: str, graphs_uris: List[str], schemas: List[str], output: Path) -> str:
```
Comment (costezki) [inline validator/service_layer/handlers.py:66]: commented code will most likely be deleted in the future

**Review by cosletmihai — APPROVED** [review]
current implementation of the report generation gets the `.ttl` from RDFUnit and then passes it to `eds4jijnja2` to get a `.html` report which is eventually sent as the api response?

File: validator/service_layer/handlers.py [hunk 6]
```diff
@@ -138,6 +154,19 @@ def generate_validation_report(path_to_report: Path, output: Path) -> None:
...
     """
 
-    report_builder = ReportBuilder(path_to_report, output)
+    report_builder = ReportBuilder(path_to_report)
     report_builder.add_after_rendering_listener(__copy_static_content)
     report_builder.make_document()
+    return str(path_to_report) + "/output/" + "main.html"
+
+
+def prepare_eds4jinja_context(report_path, source_file):
```
Comment (costezki) [inline validator/service_layer/handlers.py:163]: so you open and update the existing file rather than creating the config from scratch. 
Fair enough

**Review by costezki — APPROVED** [review]
Well done!

**Comment by cosletmihai** [comment]
@mclaurentiu, put an `_` in front of the **failing** tests so that they are not run (example: `def _test_this_will_not_run()`) and you'll be able to merge the request.

I fixed/will fix the tests on my branch

File: validator/service_layer/handlers.py (hunk 5 above)
Comment (cosletmihai) [reply validator/service_layer/handlers.py:66]: I was thinking the same. 

will we need to have this type of validation eventually?

File: validator/service_layer/handlers.py (hunk 6 above)
Comment (mclaurentiu) [reply validator/service_layer/handlers.py:163]: The entire template structure is copied to the temporary output folder, the configuration file is adjusted accordingly, eds2jinja4 runs, and after the generated report is returned, the temporary folder is removed.

**Review by costezki — APPROVED** [review]

File: validator/service_layer/handlers.py (hunk 5 above)
Comment (costezki) [reply validator/service_layer/handlers.py:66]: most likely, not

**Pull request merged**